src/corehost/inc/pal.h

```diff
@@ -96,7 +98,7 @@ namespace pal
 
     bool realpath(string_t& path);
     bool file_exists(const string_t& path);
-    std::vector<pal::string_t> readdir(const string_t& path);
+    void readdir(const string_t& path, std::vector<pal::string_t>* list);
```

Comment:
I am confused - we are using reference variable for path, why can we not use reference for list?